src/main/kotlin/uk/gov/dwp/dataworks/logging/LogFields.kt

```diff
@@ -0,0 +1,59 @@
+package uk.gov.dwp.dataworks.logging
+
+import java.net.InetAddress
+
+/**
+ * Object to hold the Application specific and Common fields that should be added to each of the log lines output by
+ * this library. Provides helper methods to return these fields as a set of json compliant Key:value pairs but without
+ * any wrapping (i.e no `{}`).
+ */
+object LogFields {
+    private val logFields = CommonLogFields.commonFields.toMutableMap()
+    val asJson = logFields.map { """ "${it.key}":"${it.value}" """.trim() }.joinToString(separator = ",")
+
+    fun get(logField: String): String {
+        return logFields.getValue(logField)
+    }
+
+    fun get(logField: LogField): String {
+        return get(logField.systemPropName)
+    }
+
+    fun put(logField: String, value: String) {
+        if(logField.isBlank() || value.isBlank()) {
+            throw IllegalArgumentException("Key and value must not be blank! Got: $logField:$value")
+        }
+        logFields[logField] = value
+    }
+}
+
+/**
+ * Object which represents fields that are common across all applications. These are extracted from Environment or
+ * Java System variables and used to prepend all log lines sent by this library.
+ *
+ * Variables are resolved from the following places, in preferential order:
+ * * Environment variable - [System.getenv]
+ * * System property - [System.getProperty]
+ * * [LogField.default]
+ */
+object CommonLogFields {
+    val commonFields: Map<String, String> =
+        LogField.values().associate {
+            val value = resolveFieldValue(it)
+            Pair(it.systemPropName, value)
+        }
+
+    private fun resolveFieldValue(logField: LogField): String {

```

Comment:
This would be ideal to expose to the user, next to put() and calling put(), so we have an `addProperty("ENV_VAR", "system.property", "default-value")` method, so the user can keep parity with what the code can do

Reply:
Have a look at my resolution for [this comment](https://github.com/dwp/dataworks-common-logging/pull/1#discussion_r390197741). I have added functionality to `put()` which hopefully meets the use case described here. Let me know your thoughts.